python/ray/serve/multiplex.py

```diff
@@ -90,8 +90,49 @@ def __init__(
         # Whether to push the multiplexed replica info to the controller.
         self._push_multiplexed_replica_info: bool = False
 
-        # Push the model IDs to the controller periodically.
-        run_background_task(self._push_model_ids())
+        # Model cache lock
+        self._model_cache_lock = asyncio.Lock()
+        self._model_load_tasks: Set[str] = set()
+
+        self.metrics_pusher = MetricsPusher()
+        self.metrics_pusher.register_task(
+            self._push_model_ids_info,
+            PUSH_MULTIPLEXED_MODEL_IDS_INTERVAL_S,
+        )
+        self.metrics_pusher.start()
+
+    def _get_model_ids(self) -> List[str]:
+        """Get the model IDs of the loaded models & loading models in the replica."""
+        models_list = set(self.models.keys())
+        models_list.update(self._model_load_tasks)
+        return list(models_list)
+
+    def _push_model_ids_info(self):
+        """Push the multiplexed replica info to the controller."""
+        try:
+            if self._push_multiplexed_replica_info:
+                get_global_client().record_multiplexed_replica_info(
+                    MultiplexedReplicaInfo(
+                        self._deployment_name,
+                        self._replica_tag,
+                        self._get_model_ids(),
+                    )
+                )
+                self._push_multiplexed_replica_info = False
+        except Exception as e:
+            logger.warning(
+                "Failed to push the multiplexed replica info "
+                f"to the controller. Error: {e}"
+            )
+
+    def __del__(self):
+        while len(self.models) > 0:
+            try:
+                asyncio.run(self.unload_model())
```

Comment:
hmm I don't think this works -- this will error if the asyncio loop is already running

Reply:
Nice catch, I fool myself by checking the unit test, looks like test can ignore exception. Updated the replica.py to make sure we call the del for the multiplexed. and added a test to make sure it is called when replica terminating.